### DISCUSSION TITLE
Execute rules which needs inputs from distinct EdgeX events

### DISCUSSION THREAD

**Discussion by BlancoMY** [lf-edge/ekuiper#2269] [Q&A] [answered]
Hello,

How can we configure eKuiper to execute a rule that combines inputs from distinct EdgeX events? For example, we have two EdgeX devices: device 1, which collects current data every 3 seconds, and device 2, which collects voltage data every 5 seconds. Is it possible to create an eKuiper rule to calculate power (current * voltage) every 3 seconds by merging these disparate event streams?

**Comment by ngjaying** [comment, marked as answer]
Hello, 
Please check cookbook https://ekuiper.org/docs/en/latest/example/data_merge/merge_single_stream.html. Solution 2 may be what you need?

**Reply by BlancoMY** [reply]
Thank You so much
